### ISSUE TITLE
prefer-conditional-expression: checks deeply nested else if

### ISSUE THREAD

**Issue by ajafff** [palantir/tslint#3350]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
if (length === 0) {
    result = "foo";
} else if (length === 1) {
    result = "bar";
} else if (length === 2) { // error on this line
    result = "something else";
} else {
    result = "unknown";
}
```

with `tslint.json` configuration:

```json
"prefer-conditional-expression": true
```

#### Actual behavior

Error for the last `if else` nested inside `if else if else`. Note that the `if else` count needs to be even for this behavior to occur.

#### Expected behavior

No error, because `"check-else-if"` is not enabled.

**Comment by adidahiya** [comment]
let's try to get this in v5.9
